server/etcdserver/server.go

```diff
@@ -1163,10 +1163,8 @@ func (s *EtcdServer) NewUberApplier() apply.UberApplier {
 }
 
 func verifySnapshotIndex(snapshot raftpb.Snapshot, cindex uint64) {
-	verify.Verify(func() {
-		if cindex != snapshot.Metadata.Index {
-			panic(fmt.Sprintf("consistent_index(%d) isn't equal to snapshot index (%d)", cindex, snapshot.Metadata.Index))
-		}
+	verify.VerifyPrototype(func() (bool, string, map[string]any) {
+		return cindex != snapshot.Metadata.Index, fmt.Sprintf("consistent_index(%d) isn't equal to snapshot index (%d)", cindex, snapshot.Metadata.Index), nil
```

Comment:
I'm thinking that it would be easier to use static string and only allow passing dynamic values via details.